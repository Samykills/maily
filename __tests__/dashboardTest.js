/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import "react-native";
import React from "react";
import Dashboard from "../app/views/dashboard/dashboard";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
//mock the hide function of splashscreen

jest.useFakeTimers();
jest.mock("react-native-splash-screen", () => ({
  hide: jest.fn()
}));
it("Dashboard renders correctly", () => {
  const tree = renderer.create(<Dashboard />).toJSON();
  expect(tree).toMatchSnapshot();
});
