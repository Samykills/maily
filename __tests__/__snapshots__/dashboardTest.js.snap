// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "flex": 2,
        },
      ]
    }
  >
    <View
      header={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": null,
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "#d1d8e0",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 28,
                "width": 35,
              },
              Object {
                "color": "#4b7bec",
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        type="MaterialCommunityIcons"
      >
        ï‡®
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "color": "#000000BC",
              "fontFamily": undefined,
              "fontSize": 27.546799160701053,
              "fontWeight": "500",
            },
          ]
        }
        uppercase={false}
      >
        Inbox
      </Text>
    </View>
    <View
      cardBody={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": -5,
            "paddingLeft": -5,
            "paddingRight": -5,
            "paddingTop": -5,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <RCTScrollView
        ListEmptyComponent={
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Styled(Text)
              note={true}
              style={
                Object {
                  "fontFamily": undefined,
                  "fontSize": 22.955665967250873,
                }
              }
            >
              You do not have any mails as yet!
            </Styled(Text)>
          </View>
        }
        ListHeaderComponent={null}
        data={Array []}
        disableVirtualization={false}
        extraData={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                null,
              ]
            }
          >
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginTop": 5,
                  },
                  Object {
                    "fontFamily": undefined,
                    "fontSize": 22.955665967250873,
                  },
                ]
              }
              uppercase={false}
            >
              You do not have any mails as yet!
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00000060",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomWidth": 0,
              "borderColor": "#ccc",
              "borderLeftWidth": 0,
              "borderRadius": 2,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "elevation": null,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginLeft": 135,
              "marginRight": 135,
              "transform": Array [
                Object {
                  "translateY": 400.2,
                },
              ],
            },
          ]
        }
        transparent={true}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#eb3b5a"
              hidesWhenStopped={true}
              size="large"
              style={
                Object {
                  "height": 80,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#eb3b5a",
                  "fontFamily": "Roboto",
                  "fontSize": 30.607554623001164,
                  "fontWeight": "300",
                  "marginLeft": 22.5,
                }
              }
            >
              Syncing Mail...
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      header={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": null,
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "#d1d8e0",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 28,
                "width": 35,
              },
              Object {
                "color": "#4b7bec",
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        type="MaterialCommunityIcons"
      >
        ï‡®
      </Text>
      <Text
        style={
          Object {
            "color": "#000000BC",
            "fontFamily": undefined,
            "fontSize": 27.546799160701053,
            "fontWeight": "500",
          }
        }
      >
        Content
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000000BC",
            "fontFamily": undefined,
            "fontSize": 27.546799160701053,
            "fontWeight": "500",
          }
        }
      >
        Click on mails above to view the content
      </Text>
    </View>
  </View>
</View>
`;
